### PR TITLE
Added client notification popups for new/deleted orders.

### PR DESCRIPTION
**Pull request by KPull**
Added popup notifications when a new order is created/closed.

Need to configure:
in web.js: set the environment variable called IOPORT which is the port number of websocket connections.
in apiPrefix.js: set the URL for the websocket endpoint. Use the same port used for the IOPORT env variable above.

### CODE REVIEW

**Comment by camilleriluke** [comment]
:sweet_potato: awesome!!!

File: webapp/app/scripts/app.js [hunk 1]
```diff
@@ -28,4 +30,41 @@ angular.module('ikelClientApp', [
...
+            }
+        });
+    });
+    ehnSocket.on('closed_order', function(order) {
+        webNotification.showNotification('Food Order Closed', {
+            body: 'The food order for ' + order.from.name + ' by ' + order.author + ' has been closed.',
+            icon: 'images/burger.png',
+            autoClose: 24000
+        }, function(error, hide) {
+            if (error) {
```
Comment (camilleriluke) [inline webapp/app/scripts/app.js]: Do we really need this callback?

File: webapp/app/scripts/app.js [hunk 2]
```diff
@@ -28,4 +30,41 @@ angular.module('ikelClientApp', [
...
+    });
+    
+    ehnSocket.on('new_order', function(order) {
+        webNotification.showNotification('New Food Order', {
+            body: order.author + ' has opened a new food order for ' + order.from.name + ' on Ejja Ħa Nieklu.',
+            icon: 'images/burger.png',
+            autoClose: 24000
+        }, function(error, hide) {
+            if (error) {
+                console.log('Unable to show notification ', error);
```
Comment (camilleriluke) [inline webapp/app/scripts/app.js]: Do we really need this callback?

File: webapp/app/scripts/services/apiPrefix.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
 'use strict';
 
-angular.module('ikelClientApp').constant('apiPrefix', 'http://ejja-ha-nieklu.herokuapp.com');
+angular.module('ikelClientApp').constant('apiPrefix', 'http://localhost:5000');
```
Comment (camilleriluke) [inline webapp/app/scripts/services/apiPrefix.js]: `localhost`??

We need to find a better solution how to abstract this. 

Maybe by setting up a local proxy which will route all the traffic to the local server during dev mode.

File: webapp/app/scripts/app.js (hunk 1 above)
Comment (KPull) [inline webapp/app/scripts/app.js]: Not really. I had it there for debugging purposes but I'll remove it.

File: webapp/app/scripts/services/apiPrefix.js (hunk 3 above)
Comment (KPull) [inline webapp/app/scripts/services/apiPrefix.js]: I can't think of anything for this but if I see something I'll tell you. In the meantime, I'll fix this to its original state.

**Comment by KPull** [comment]
By the way, I didn't check to see how this all works on non-Firefox browsers.

**Comment by camilleriluke** [comment]
You might need to merge the latest master into your fork.

**Comment by KPull** [comment]
Completed changes.

File: package.json [hunk 4]
```diff
@@ -10,6 +10,7 @@
   "author": "",
   "license": "BSD-2-Clause",
   "dependencies": {
+    "socket.io": "^1.3.7",
```
Comment (camilleriluke) [inline package.json]: I would be idea if we start using `=` instead of `^`.

I'll update it myself after the merge

File: web.js [hunk 5]
```diff
@@ -8,6 +8,9 @@ var cors = require('cors');
 var bodyParser = require('body-parser');
 var nodemailer = require('nodemailer');
 
+var ioPort = Number(process.env.IOPORT || 3000);
```
Comment (camilleriluke) [inline web.js]: I don't see why we need this on a different port. Wouldn't it be the same if we use a different url?

Also, is there any reason why we need the port as an env. variable?

Comment (KPull) [inline web.js]: I did try using the same port but as far as I know it doesn't let you (you get two TCP connections on one port).

You would need to find a replacement for Express that can handle both standard HTTP requests and websocket requests on a single port.

Comment (camilleriluke) [inline web.js]: Might be useful http://socket.io/docs/

**Comment by KPull** [comment]
Changed code to have socket.io use the same server host as the Express app.

**Comment by camilleriluke** [comment]
Tried to merge PR but site failed to load after `grunt build`. Please make sure that injected modules observe the obfuscation-safe pattern.

**Comment by KPull** [comment]
I'll look into it.

Kyle


Sent from my Samsung Galaxy smartphone.


-------- Original message --------

**Pull request merged**